log4jSyslogWriter64k/src/main/java/com/github/loggly/log4j/helpers/SyslogTcpWriter64k.java

```diff
@@ -70,5 +76,8 @@ public void close() throws IOException {
 		if (socket.isPresent()) {
 			socket.get().close();
 		}
+
+		writer = Optional.empty();
+		socket = Optional.empty();
```

Comment:
Please do this in a finally block as there might occur additional IOExceptions in error cases.